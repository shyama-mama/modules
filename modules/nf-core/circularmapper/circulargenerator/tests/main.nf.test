// nf-core modules test circularmapper/circulargenerator
nextflow_process {

    name "Test Process CIRCULARMAPPER_CIRCULARGENERATOR"
    script "../main.nf"
    process "CIRCULARMAPPER_CIRCULARGENERATOR"
    config "./nextflow.config"

    tag "modules"
    tag "modules_nfcore"
    tag "circularmapper"
    tag "circularmapper/circulargenerator"

    test("sarscov2 - fasta") {

        when {
            process {
                """
                input[0] = [
                            [ id:'test' ],
                            file(params.test_data['sarscov2']['genome']['genome_fasta'], checkIfExists: true)
                    ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
                //TODO nf-core: Add all required assertions to verify the test output.
                // See https://nf-co.re/docs/contributing/tutorials/nf-test_assertions for more information and examples.
            )
        }

    }


}
