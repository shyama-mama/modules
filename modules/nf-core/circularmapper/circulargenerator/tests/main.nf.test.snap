{
    "sarscov2 - fasta": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "genome_500.fasta:md5,2e454d6187b10dfe43e67b121e099ab6"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test"
                        },
                        "genome.fasta_500_elongated:md5,3a9ea6d336e113a74d7fdca5e7b623fc"
                    ]
                ],
                "2": [
                    "versions.yml:md5,48802d9192befc5321e734a143487334"
                ],
                "contig": [
                    [
                        {
                            "id": "test"
                        },
                        "genome.fasta_500_elongated:md5,3a9ea6d336e113a74d7fdca5e7b623fc"
                    ]
                ],
                "fasta": [
                    [
                        {
                            "id": "test"
                        },
                        "genome_500.fasta:md5,2e454d6187b10dfe43e67b121e099ab6"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,48802d9192befc5321e734a143487334"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.04.1"
        },
        "timestamp": "2024-03-20T09:53:03.061512"
    }
}